cpp/tests/test_odesecir.cpp

```diff
@@ -1177,5 +1178,206 @@ TEST(Secir, read_population_data_one_age_group)
+                MatrixNear(print_wrap(expected_results[0].get_groups().matrix()), 1e-5, 1e-5));
+}
+
+TEST(TestOdeSecir, export_time_series_init_old_date)
+{
+    TempFileRegister temp_file_register;
+    auto tmp_results_dir = temp_file_register.get_unique_path();
+    ASSERT_THAT(mio::create_directory(tmp_results_dir), IsSuccess());
+
+    constexpr auto num_age_groups = 6; //reading data requires RKI data age groups
+    mio::osecir::Model<double> model(make_model(num_age_groups));
+
+    // Test exporting time series
+    std::vector<mio::osecir::Model<double>> models{model};
+    ASSERT_THAT(mio::osecir::export_input_data_county_timeseries(
+                    models, tmp_results_dir, {1002}, {1000, 12, 01}, std::vector<double>(size_t(num_age_groups), 1.0),
+                    1.0, 0, mio::path_join(TEST_DATA_DIR, "county_divi_ma7.json"),
+                    mio::path_join(TEST_DATA_DIR, "cases_all_county_age_ma7.json"),
+                    mio::path_join(TEST_DATA_DIR, "county_current_population.json")),
+                IsSuccess());
+
+    auto data_extrapolated = mio::read_result(mio::path_join(tmp_results_dir, "Results_rki.h5"));
+    ASSERT_THAT(data_extrapolated, IsSuccess());
+    auto results_extrapolated = data_extrapolated.value()[0].get_groups().get_value(0);
+
+    // if we enter an old date, the model only should be initialized with the population data.
+    // read population data
+    std::string path = mio::path_join(TEST_DATA_DIR, "county_current_population.json");
+    const std::vector<int> region{1002};
+    auto population_data = mio::osecir::details::read_population_data(path, region, false).value();
+
+    // So, the expected values are the population data in the susceptible compartments and zeros in the other compartments.
+    for (auto i = 0; i < num_age_groups; i++) {
+        EXPECT_EQ(results_extrapolated(i * Eigen::Index(mio::osecir::InfectionState::Count)), population_data[0][i]);
+    }
+    // sum of all compartments should be equal to the population
+    EXPECT_EQ(results_extrapolated.sum(), std::accumulate(population_data[0].begin(), population_data[0].end(), 0.0));
+}
+
+// // Model initialization should return same start values as export time series on that day
+TEST(TestOdeSecir, model_initialization)
+{
+    constexpr auto num_age_groups = 6; //reading data requires RKI data age groups
+    mio::osecir::Model<double> model(make_model(num_age_groups));
+
+    // Vector assignment necessary as read_input_data_county changes model
+    auto model_vector = std::vector<mio::osecir::Model<double>>{model};
+
+    ASSERT_THAT(mio::osecir::read_input_data_county(model_vector, {2020, 12, 01}, {1002},
+                                                    std::vector<double>(size_t(num_age_groups), 1.0), 1.0,
+                                                    TEST_DATA_DIR, 2, false),
+                IsSuccess());
+
+    // Values from data/export_time_series_init_osecir.h5, for reading in comparison
```

Comment:
Wouldn't it be better to really use (the first entry of) the data from the h5 file instead of the copy? If something is changed in the h5 file, one may not think of changing these lines also, which could lead to inconsistencies between export... and read...

Reply:
i would leave stay with the values here, since the read_input_data_county is not directly correlated with the h5 file